tests/vcs/git/GitCommitBearTest.py

```diff
@@ -139,6 +139,17 @@ def test_shortlog_checks_shortlog_trailing_period(self):
         self.assertEqual(self.run_uut(shortlog_trailing_period=False), [])
         self.assertEqual(self.run_uut(shortlog_trailing_period=None), [])
 
+    def test_shortlog_checks_imperative(self):
+        self.git_commit("tag: Add shortlog in imperative")
+        self.assertNotIn("Shortlog of HEAD commit isn't imperative mood",
+                         ''.join(self.run_uut()))
+        self.git_commit("Added invalid shortlog")
+        self.assertIn("Shortlog of HEAD commit isn't imperative mood",
+                      ''.join(self.run_uut()))
+        self.git_commit("Adding another invalid shortlog")
+        self.assertIn("Shortlog of HEAD commit isn't imperative mood",
+                      ''.join(self.run_uut()))
```

Comment:
I think maybe we want to either document or find solutions to the things it
cannot do at the very least


On Sat, Apr 2, 2016 at 7:35 PM, Tushar <notifications@github.com> wrote:

> In tests/vcs/git/GitCommitBearTest.py
> <https://github.com/coala-analyzer/coala-bears/pull/315#discussion_r58293799>
> :
>
> > @@ -139,6 +139,17 @@ def test_shortlog_checks_shortlog_trailing_period(self):
> >          self.assertEqual(self.run_uut(shortlog_trailing_period=False), [])
> >          self.assertEqual(self.run_uut(shortlog_trailing_period=None), [])
> >
> > +    def test_shortlog_checks_imperative(self):
> > +        self.git_commit("tag: Add shortlog in imperative")
> > +        self.assertNotIn("Shortlog of HEAD commit isn't imperative mood",
> > +                         ''.join(self.run_uut()))
> > +        self.git_commit("Added invalid shortlog")
> > +        self.assertIn("Shortlog of HEAD commit isn't imperative mood",
> > +                      ''.join(self.run_uut()))
> > +        self.git_commit("Adding another invalid shortlog")
> > +        self.assertIn("Shortlog of HEAD commit isn't imperative mood",
> > +                      ''.join(self.run_uut()))
>
> So IMO we can find many cases that'd fail the check. I am trying to cover
> the most common structures. Hm considering the dilemma the problem requires
> more thoughts maybe? Maybe an overkill? :P
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/coala-analyzer/coala-bears/pull/315/files/086478f59e0ea256e3329e7bb1047f53e55c9e75#r58293799>
>